content/posts/aws-autoscaling-groups-with-terraform.md

```diff
@@ -0,0 +1,193 @@
+---
+title: "Updating AWS autoscaling groups with Terraform and preserve desired capacity"
+date: 2020-06-21
+authors:
+  - author:
+      name: "Ariel Peltz"
+      link: "https://github.com/arielpeltz"
+tags: ["aws", "terraform", "autoscaling", "desired_capacity"]
+---
+
+## Introduction
+Terraform is probably the best tool for deploying infrastructure.  So it was an obvious choice
+when we wanted to deploy an autoscaling group into AWS.  The project we are working on requires us to
+deploy a service on instances in AWS.  The service is stateless and has simple configuration that is
+easy to configure using [cloud-init](https://cloudinit.readthedocs.io/en/latest/index.html).
+
+Our intention is to be able to configure everything with Terraform. Every time we need to make
+a configuration change, we update Terraform configuration and apply again.
+We wanted to have clean instances when we make any configuration change and not update in place.
+
+> Throughout the examples we omitted some configuration elements creation for brevity.
+> Others are injected as variables and are also not shown here.
+
+## Initial Configuration
+First create the launch template
```

Comment:
Add a colon at the end (looks like a truncated sentence otherwise)

Reply:
Fixed